applications/asset_tracker/src/main.c

```diff
@@ -1113,30 +1235,67 @@ static void work_init(void)
 /**@brief Configures modem to provide LTE link. Blocks until link is
  * successfully established.
  */
-static void modem_configure(void)
+static int modem_configure(void)
 {
 #if defined(CONFIG_BSD_LIBRARY)
 	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
 		/* Do nothing, modem is already turned on
 		 * and connected.
 		 */
-	} else {
-		int err;
-
-		LOG_INF("Connecting to LTE network. ");
-		LOG_INF("This may take several minutes.");
-		ui_led_set_pattern(UI_LTE_CONNECTING);
-
-		err = lte_lc_init_and_connect();
+		goto connected;
+	}
+
+	ui_led_set_pattern(UI_LTE_CONNECTING);
+	LOG_INF("Connecting to LTE network.");
+	LOG_INF("This may take several minutes.");
+
+#if defined(CONFIG_LWM2M_CARRIER)
+#if !defined(CONFIG_GPS_USE_SIM)
+/* Usually the modem configuration is done during boot, */
+/* but when LWM2M carrier library is enabled, BSD library */
+/* is not enabled before now, leaving the configuration */
+/* to the application. */
+#if defined(CONFIG_BOARD_NRF9160_PCA20035NS)
+	const char *const cmds[] = {
+			  "AT%XMAGPIO=1,1,1,7,1,746,803,2,698,748,"
+			  "2,1710,2200,3,824,894,4,880,960,5,791,849,"
+			  "7,1574,1577",
+			  "AT%XMODEMTRACE=0"};
+#elif defined(CONFIG_BOARD_NRF9160_PCA10090NS)
+	const char *const cmds[] = {
+			  "AT\%XMAGPIO=1,0,0,1,1,1574,1577",
+			  "AT\%XCOEX0=1,1,1570,1580"};
```

Comment:
I think we should try to find a better way to handle this COEX and MAGPIO stuff, and hide it from the app. Not sure how and where.
Perhaps link controller is a better place for it, even though it doesn't concern the actual LTE link? 
@bjda @jhn-nordic Any thoughts?

Reply:
Could it be possible to implement this as a weak function in link_controller and then have implementations in the board folders when needed?

Reply:
i agree, it would be nice to get this out of the app.

Reply:
I think we should put it into `w_lte_lc_init()` in `lte_lc.c`, at least for now.

Reply:
updated.